common/utils.go

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-mysql-org/go-mysql/canal"
+	"github.com/jmoiron/sqlx"
+)
+
+func FormatValue(val any) any {
+	switch val := val.(type) {
+	case time.Time:
+		return val.UTC().Format(time.RFC3339)
+	case *time.Time:
+		return val.UTC().Format(time.RFC3339)
+	case int8:
+		return int(val)
+	case int16:
+		return int(val)
+	case int32:
+		return int(val)
+	case int64:
+		return int(val)
+	case uint8:
+		return int(val)
+	case uint16:
+		return int(val)
+	case uint32:
+		return int(val)
+	case uint64:
+		return int(val)
+	case float32:
+		return float64(val)
+	case float64:
+		return val
+	case []uint8:
+		return string(val)
+	default:
+		return val
+	}
+}
+
+func NewSqlxDB(config Config) (*sqlx.DB, error) {
+	dataSourceName := fmt.Sprintf(
```

Comment:
Rather than using `Sprintf` perhaps using [`mysql.Config`](https://github.com/go-sql-driver/mysql/blob/master/dsn.go#L37) for the driver and then passing `config.FormatDSN()` to the sqlx.

Reply:
While I agree with you that that indeed is a better solution, I've got another pr here #10 that changes the source configuration, so that it is more similar to the postgres connector. I'd just leave this as it is.

Reply:
Makes sense, URL should be enough.